test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua

```diff
@@ -0,0 +1,72 @@
+local net = require('net.box')
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.before_each(function(cg)
+    cg.server:exec(function()
+        box.schema.create_space('test', {
+            id = 621,
+            format = {
+                {name = "id", type = "unsigned"},
+                {name = "name", type = "string"},
+            },
+        })
+        box.space.test:create_index('primary', {
+            type = 'tree',
+            unique = true,
+            parts = {1, 'unsigned'},
+        })
+        box.space.test:create_index('secondary', {
+            id = 5,
+            type = 'tree',
+            unique = true,
+            parts = {2, 'string'},
+        })
+    end)
+    cg.conn = net.connect(cg.server.net_box_uri, {fetch_schema = false})
+    t.assert_is_not(g.conn, nil)
+end)
+
+g.after_each(function()
+    g.conn:close()
+    g.server:exec(function()
+        if box.space.test then
+            box.space.test:drop()
+        end
+    end)
+end)
+
+g.test_index_select = function(cg)
+    local space = cg.conn.space.test
+    t.assert_is_not(space, nil)
+
+    local res1 = space.index.secondary:select()
+    local res2 = space.index[5]:select()
+    t.assert_equals(res1, res2)
```

Comment:
Let's insert some tuples to the space along the way - there is no point to compare results since they are empty.